File: frontend/Product.swift
//
//  Product.swift
//  GTThriftShop
//
//  Created by Mengyang Shi on 2/17/17.
//  Copyright © 2017 Triple6. All rights reserved.
//

import UIKit

class Product: NSObject, NSCoding {
    var imageUrls: [String]!
    let name: String!
    let price: String!
    let info: String!
    let pid: Int!
    let postTime: String!
    let usedTime: String!
    let userId: Int!
    let userName: String!
    var isSold: Bool!
    
    init(name: String, price: String, info: String, pid: Int, postTime: String, usedTime: String, userId: Int, userName: String, imageUrls: [String], isSold: Bool) {
        self.name = name
        self.price = price
        self.info = info
        self.pid = pid
        self.postTime = postTime
        self.usedTime = usedTime
        self.userId = userId
        self.userName = userName
        self.imageUrls = imageUrls
        self.isSold = isSold
    }
    
    required init(coder decoder: NSCoder) {
        self.name = decoder.decodeObject(forKey: "name") as! String
        self.price = decoder.decodeObject(forKey: "price") as! String
        self.info = decoder.decodeObject(forKey: "info") as! String
        self.pid = decoder.decodeInteger(forKey: "pid")
        self.postTime = decoder.decodeObject(forKey: "postTime") as! String
        self.usedTime = decoder.decodeObject(forKey: "usedTime") as! String
        self.userId = decoder.decodeInteger(forKey: "userId")
        self.userName = decoder.decodeObject(forKey: "userName") as! String
        self.imageUrls = decoder.decodeObject(forKey: "imageUrls") as! [String]
        self.isSold = decoder.decodeBool(forKey: "isSold")
    }
    
    func encode(with aCoder: NSCoder) {
        aCoder.encode(name, forKey: "name")
        aCoder.encode(price, forKey: "price")
        aCoder.encode(info, forKey: "info")
        if let pid = pid {
            aCoder.encode(pid, forKey: "pid")
        }
        aCoder.encode(postTime, forKey: "postTime")
        aCoder.encode(usedTime, forKey: "usedTime")
        if let userId = userId {
            aCoder.encode(userId, forKey: "userId")
        }
        aCoder.encode(userName, forKey: "userName")
        aCoder.encode(imageUrls, forKey: "imageUrls")
        if let isSold = isSold {
            aCoder.encode(isSold, forKey: "isSold")
        }
        
    }
    
    func setSold() {
        if self.isSold == false {
            self.isSold = true
        }
    }
}

File: android/GTThriftShop/app/src/main/java/triple6/gtthriftshop/models/Product.java
package triple6.gtthriftshop.models;

import java.util.ArrayList;

/**
 * Product.java
 * GTThriftShop
 *
 * Created by Wenzhong Jin on 7/16/17.
 * Copyright © 2017 Triple6. All rights reserved.
 */

class Product {
    private int userId;
    private String userName;
    private int pid;
    private String pName;
    private double pPrice;
    private String pInfo;
    private String postTime;
    private String usedTime;
    private boolean isSold;
    private ArrayList<String> imageURLs;

    public Product(int userId, String userName, int pid, String pName, double pPrice, String pInfo, String postTime,
                   String usedTime, boolean isSold, ArrayList<String> imageURLs) {
        this.userId = userId;
        this.userName = userName;
        this.pid = pid;
        this.pName = pName;
        this.pPrice = pPrice;
        this.pInfo = pInfo;
        this.postTime = postTime;
        this.usedTime = usedTime;
        this.isSold = isSold;
        this.imageURLs = imageURLs;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public int getPid() {
        return pid;
    }

    public void setPid(int pid) {
        this.pid = pid;
    }

    public String getpName() {
        return pName;
    }

    public void setpName(String pName) {
        this.pName = pName;
    }

    public double getpPrice() {
        return pPrice;
    }

    public void setpPrice(double pPrice) {
        this.pPrice = pPrice;
    }

    public String getpInfo() {
        return pInfo;
    }

    public void setpInfo(String pInfo) {
        this.pInfo = pInfo;
    }

    public String getPostTime() {
        return postTime;
    }

    public void setPostTime(String postTime) {
        this.postTime = postTime;
    }

    public String getUsedTime() {
        return usedTime;
    }

    public void setUsedTime(String usedTime) {
        this.usedTime = usedTime;
    }

    public boolean isSold() {
        return isSold;
    }

    public void setSold(boolean sold) {
        isSold = sold;
    }

    public ArrayList<String> getImageURLs() {
        return imageURLs;
    }

    public void setImageURLs(ArrayList<String> imageURLs) {
        this.imageURLs = imageURLs;
    }
}
